kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTable.js

```diff
@@ -391,22 +434,56 @@ define([
                 e.target.disabled = true;
                 return false;
             }
+            // make sure any errors associated with the row ID are deleted
+            if (errors[target]) {
+                delete errors[target];
+            }
             jobManager.doJobAction(action, [target]);
 
             // if the job is being retried and the log viewer is open, close it
             // TODO: a better fix!
-            if (
-                action === 'retry' &&
-                $(e.target).closest('tr')[0].classList.contains('vertical_collapse--open')
-            ) {
-                showHideChildRow(e);
+            if (action === 'retry') {
+                closeRow(e);
             }
 
             // disable the button to prevent further clicks
             e.target.disabled = true;
+            e.target.textContent = action + 'ing';
+            // remove any error indicator from the DOM
+            $(e.target).closest('td').find('.kb-job-status__icon--action_warning').remove();
             return true;
         }
 
+        /**
+         * add the listeners to the table that allow row expansion
+         */
+        function addTableClickListeners() {
```

Comment:
extracted this out into its own function as it is needed if new rows are added to the table